src/datatype.c

```diff
@@ -689,8 +689,9 @@ static jl_value_t *boxed_char_cache[128];
 JL_DLLEXPORT jl_value_t *jl_box_char(uint32_t x)
 {
     jl_ptls_t ptls = jl_get_ptls_states();
-    if (0 < (int32_t)x)
-        return boxed_char_cache[x >> 24];
+    uint32_t u = __builtin_bswap32(x);
```

Comment:
We have `bswap_32` defined for this for a few platforms.

Reply:
"for a few platforms" sounds unlikely to work everywhere. @yuyichao, @Keno, @vtjnash, do any of you have advice on how to spell this operation correctly and portably?

Reply:
`bswap_32` is the correct version since it takes care of the difference between clang, ICC, MSVC, or not having the builtin available at all.

see https://github.com/JuliaLang/julia/blob/75075114d407eb5d4bf1604880e3423052d96887/src/support/utils.h#L32-L57